File: MiniProject/SingletonState.java
package miniProject;

import miniProject.StateName.PageName;
import miniProject.StateName.UserType;

public class SingletonState {

	private static SingletonState instance = null;
	public UserType currentUserType = null;
	public PageName currentPage = PageName.LOGIN_PAGE;
	public static ProductManager productManager;
	
	public final static SingletonState getState() {
		if (SingletonState.instance == null) {
			SingletonState.instance = new SingletonState();
			productManager = new ProductManager();
		}
		return instance;
	}
	
	public final ProductManager getProductManager() {
		return productManager;
	}
	

}

File: MiniToTypeScript/singletonState.js
"use strict";
exports.__esModule = true;
exports.SingletonState = void 0;
var productManager_js_1 = require("./productManager.js");
var myEnum_js_1 = require("./myEnum.js");
var SingletonState = /** @class */ (function () {
    function SingletonState() {
    }
    SingletonState.prototype.getState = function () {
        if (this.instance == null) {
            this.instance = new SingletonState();
            this.productManager = new productManager_js_1.ProductManager();
            this.currentPage = myEnum_js_1.PageName.LOGIN_PAGE;
        }
        return this.instance;
    };
    SingletonState.prototype.getProductManager = function () {
        return this.productManager;
    };
    return SingletonState;
}());
exports.SingletonState = SingletonState;
